youtube_dl/extractor/tvnoe.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import re
+from youtube_dl.compat import compat_urlparse
+from .common import InfoExtractor
+from ..utils import clean_html, get_element_by_class
+
+SRC_PATTERN = r'<script.+jwplayer.*sources:\s*(\[[^]]+\])'
+
+
+class TVNoeIE(InfoExtractor):
+    _VALID_URL = r'https?://(www\.)?tvnoe\.cz/video/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'http://www.tvnoe.cz/video/10362',
+        'md5': 'aee983f279aab96ec45ab6e2abb3c2ca',
+        'info_dict': {
+            'id': '10362',
+            'ext': 'mp4',
+            'series': 'Noční univerzita',
+            'title': 'prof. Tomáš Halík, Th.D. - ' +
+                     'Návrat náboženství a střet civilizací',
+            'description': 'md5:f337bae384e1a531a52c55ebc50fff41',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        iframe_url = self._search_regex(r'<iframe[^>]+src="([^"]+)"',
+                                        webpage, 'iframe src attribute')
+
+        ifs_page = self._download_webpage(iframe_url, video_id)
+        ifs_sources = self._search_regex(SRC_PATTERN, ifs_page, 'sources')
+        ifs_sources = json.loads(re.sub(r'(file|label):', r'"\1":',
+                                        ifs_sources))
+
+        assert len(ifs_sources) > 0
+
+        formats = [{'url': compat_urlparse.urljoin(iframe_url, source['file']),
+                    'format_id': source['label']}
+                   for source in ifs_sources]
```

Comment:
> This website uses JWPlayer. Use JWPlatformBaseIE._extract_jwplayer_data instead. Usually it's not a good practice to re-invent the wheel.

I don’t see anywhere fixing of broken JSON (the keys are not enclosed in doublequotes), something like I do in:
```python
ifs_sources = json.loads(re.sub(r'(file|label):', r'"\1":',
                                        ifs_sources))
```
which leads to this crash:
```
matej@mitmanek: youtube-dl (tvnoe *)$ python test/test_download.py TestDownload.test_TVNoe
[TVNoe] 10362: Downloading webpage
[TVNoe] 10362: Downloading webpage
ERROR: 10362: Failed to parse JSON  (caused by ValueError('Expecting property name enclosed in double quotes: line 1 column 2 (char 1)',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/common.py", line 549, in _parse_json
    return json.loads(json_string)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
Traceback (most recent call last):
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/YoutubeDL.py", line 691, in extract_info
    ie_result = ie.extract(url)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/common.py", line 347, in extract
    return self._real_extract(url)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/tvnoe.py", line 42, in _real_extract
    ifs_page, video_id, require_title=False)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/jwplatform.py", line 29, in _extract_jwplayer_data
    self._find_jwplayer_data(webpage), video_id)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/common.py", line 553, in _parse_json
    raise ExtractorError(errmsg, cause=ve)
ExtractorError: 10362: Failed to parse JSON  (caused by ValueError('Expecting property name enclosed in double quotes: line 1 column 2 (char 1)',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

E
======================================================================
ERROR: test_TVNoe (__main__.TestDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_download.py", line 141, in test_template
    force_generic_extractor=params.get('force_generic_extractor', False))
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/YoutubeDL.py", line 706, in extract_info
    self.report_error(compat_str(e), e.format_traceback())
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/YoutubeDL.py", line 565, in report_error
    self.trouble(error_message, tb)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/YoutubeDL.py", line 535, in trouble
    raise DownloadError(message, exc_info)
DownloadError: ERROR: 10362: Failed to parse JSON  (caused by ValueError('Expecting property name enclosed in double quotes: line 1 column 2 (char 1)',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

----------------------------------------------------------------------
Ran 1 test in 0.552s
```